src/sentry/static/sentry/app/views/releases/releaseCommits.jsx

```diff
@@ -29,7 +32,10 @@ const ReleaseCommit = React.createClass({
           <div className="col-xs-2"><span className="repo-label">{this.props.repository.name}</span></div>
           <div className="col-xs-2 align-right">
             <a className="btn btn-default btn-sm"
-               href={this.props.repository.url + '/' + this.props.commitId}
+               href={this.props.repository.url
+                ? (this.props.repository.url + '/' + this.props.commitId)
+                : ('http://' + this.props.repository.provider.id + '.com/' + repoPath + '/' + this.props.commitId)
+               }
                target="_blank"><span
```

Comment:
I think we should move all of this URL logic to a helper method, e.g. `getCommitURL`.

And I'd special case everything. We have no idea if Bitbucket/Stash follows the same pattern.

Something like:

```javascript
getCommitURL() {
  switch (provider) {
    case 'github':
      return url ? url + '/commit/' + sha :
        'https://github.com/' + repoPath + '/commit/' + sha;
    default:
      return url;
  }
}
```

^Something like that (totally busted).